api/src/main/java/marquez/api/NamespaceResource.java

```diff
@@ -74,8 +75,16 @@ public Response get(@PathParam("namespace") NamespaceName name) {
   public Response list(
       @QueryParam("limit") @DefaultValue("100") @Min(value = 0) int limit,
       @QueryParam("offset") @DefaultValue("0") @Min(value = 0) int offset) {
-    final List<Namespace> namespaces = namespaceService.findAll(limit, offset);
-    return Response.ok(new Namespaces(namespaces)).build();
+    final String namespaceFilter = ExclusionsFilter.getNamespacesReadFilter();
```

Comment:
To clean up the code, I suggest:

1. Define **class** `Exclusions` with a cleaner interface defined in pkg `marquez.common` (see below) in place of `ExclusionsFilter`
2. Use lambdas to call `NamespaceService.findAll()` or `NamespaceService.findAllWith()`

**`Exclusions.java`**
```java
package marquez.common;

import com.google.common.collect.ClassToInstanceMap;
import com.google.common.collect.MutableClassToInstanceMap;
import javax.annotation.Nullable;
import lombok.NonNull;

public final class Exclusions {
  private Exclusions() {}

  private static final ClassToInstanceMap<Object> EXCLUSIONS = MutableClassToInstanceMap.create();
 

  public static void use(@NonNull ExclusionsConfig config) {
    // Build exclusions map using config.
    EXCLUSIONS.put(NamespaceExclusion.class, new NamespaceExclusion(...));
  }

  public static NamespaceExclusion namespaces() {
    return EXCLUSIONS.getInstance(NamespaceExclusion.class);
  }

  public record NamespaceExclusion(@Nullable String onRead, @Nullable String onWrite) {}
}
```

Then, you can chain your calls:

**`NamespaceResource.list()`**

```java
final List<Namespace> namespaces =
  Optional.ofNullable(Exclusions.namespaces().onRead())
    .map(pattern -> namespaceService.findAllWith(pattern, limit, offset))
    .orElseGet(() -> namespaceService.findAll(limit, offset));

return Response.ok(new Namespaces(namespaces)).build();
```

Reply:
@yanlibert I'm wondering, as we work through the `Exclusions` code, would it make sense to just have:

```
exclude:
  namespaces:
     enabled: boolean
     onRead: "<pattern0>|<pattern1>|..."
     onWrite: "<pattern0>|<pattern1>|..."
```

This would give greater control and flexibility over what to exclude 😅

Reply:
I think it's a great suggestion, I refactored the code accordingly

Reply:
@wslulciuc I also think option 2 where `pattern` is a string of regex is simpler, and more straightforward to implement as I'm not a java dev 😄 
I increased the coverage and refactored the code as suggested by implementing a cleaner Exclusions interface.